hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/NSSummaryTaskDbEventHandler.java

```diff
@@ -82,23 +106,72 @@ protected void writeNSSummariesToDB(Map<Long, NSSummary> nsSummaryMap)
               e);
         }
       });
-      reconNamespaceSummaryManager.commitBatchOperation(rdbBatchOperation);
+      try {
+        reconNamespaceSummaryManager.commitBatchOperation(rdbBatchOperation);
+      } catch (IOException e) {
+        LOG.error("Failed to commit batch operation for writing NSSummary " +
+            "data in Recon DB.", e);
+        throw e;
+      }
+    }
+  }
+
+  protected void writeOrphanKeysMetaDataToDB(
+      Map<Long, OrphanKeyMetaData> orphanKeyMetaDataMap, long status)
+      throws IOException {
+    try (RDBBatchOperation rdbBatchOperation = new RDBBatchOperation()) {
+      for (Map.Entry<Long, OrphanKeyMetaData> entry :
+          orphanKeyMetaDataMap.entrySet()) {
+        try {
+          Long key = entry.getKey();
+          OrphanKeyMetaData orphanKeyMetaData =
+              orphanKeyMetaDataMap.get(key);
+          if (orphanKeyMetaData.getObjectIds().size() > 0) {
+            orphanKeyMetaData.setStatus(status);
+            reconNamespaceSummaryManager.batchStoreOrphanKeyMetaData(
+                rdbBatchOperation, key, orphanKeyMetaData);
+          } else {
+            orphanKeyMetaDataMap.remove(key);
```

Comment:
the deletion is only from map, check if deletion while iterating map works here. Need delete from table also.

Reply:
done.